docs/development/high-availability.md

```diff
@@ -11,6 +13,9 @@ spec:
     matchLabels: ...
 ```
 
+2. Capability to run with multiple replicas, e.g. by enabling leader election in case of controller.
+3. Maintain `high-availability-config.resources.gardener.cloud/type={controller, server}` label at `Deployment`s or `StatefulSet`s. If a component is a controller and (webhook) server at the same time, it may be assigned `server`.
```

Comment:
Maybe add a link to [below section](https://github.com/gardener/gardener/blob/master/docs/development/high-availability.md#convenient-application-of-these-rules) which explains this part in more detail.

Reply:
Right, I missed this section. Maybe we should add all of that information to this section? The actual goal of this PR was to keep everything that developers needs to do (like a TL;DR) in one place.

Reply:
OK, sure

Reply:
PTAL 🙂

Reply:
Didn't you also want to move the PDB part into this below section?

Reply:
Okay, done. Hope this helps developers to directly get a gist of what needs to be done from their side.